src/components/RequestStatus.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+type RequestStatusProps = {
+  requestStatus: RequestStatusOptions
+  message?: string | null
+  validationErrors?: string[] | null
+}
```

Comment:
Also make sure your props are immediately recognizable as to what they are if possible. `requestStatus` and `message` are tightly coupled in the business logic in here. But `message` doesn't immediately translate to "request response message". What if you rearranged the props like below?
```typescript
type RequestStatusProps = {
  request: {
    status: RequestStatusOptions
    message?: string | null
  }
  validationErrors?: string[] | null
}
```
This way props related to the request are logically grouped together?